### PR TITLE
Chore setup dotenv

### PR DESCRIPTION
**Pull request by thepatrickniyo**
## What this PR does

Adds dotenv setup and adds a guide to environment usage

## How to manually test it 

Create `.env` file add PORT=4000

## Env optional features

The app defines three different environments namely.

### Getting started with env 

After cloning or rebasing on main, just create three files named: **.env.development** , **.env.production** and **.env.staging**. 

Copy all the environment variables shared in **.env.example** file to each of the three files. Assign values to each variable. eg: 
`PORT=5500` in  `.env.development`
`PORT=5000` in `.env.production`
`PORT=6000` in `.env.staging`

> You can assign ports as you wish.

- **development environment**  : is the default development env that runs the app on port 5500. To be able to switch to **dev env** run. The dev environment watches changes and automatically recompile the app with new changes.

```sh
 $ npm run dev
 # yarn dev or pnpm dev
```

- **staging environment** : is the environment for testing (Running tests basically) the app before it goes into production. It runs on port 6000. To run the staging environment, use the following command:

```sh
$ npm run test
 # yarn stage or pnpm stage
```

- **production environment** : is the environment for running the app in production. It runs on port 5000. To run the production environment, use the following command:

```sh
 $ npm run start
 # yarn start or pnpm start
```

### Adding a new environment variable

This is so simple

1. To add a new environment variable, you will add it in the 3 different env files `.env.production`, `env.development` & in `.env.staging`.

2. Export the environment in the `src/config/dotenv` directory.

3. Add it to the `.env.example` without a value because `.env.example` is the only file to be pushed to github. This will aid others to updated their env files accordingly.

4. Import the variable where you want to use it from `src/config/dotenv`

4. Finally talk to the team to update them about it. 

**example** Refer to how `PORT` was set

### CODE REVIEW

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]
Change those with ts as we will use js no ts

**Comment by thepatrickniyo** [comment]
> Change those with ts as we will use js no ts

Let me work on it ASAP, thanks for the feedback

File: src/index.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import express from 'express'
+import 'dotenv/config'
+import { PORT } from './config/dotenv'
 const app = express()
-const port = process.env.PORT || 3003
```
Comment (SauveJeanLuc) [inline src/index.ts]: I think this alternative port (3003, in this case) is important and should remain.

File: README.md [hunk 2]
```diff
@@ -1,30 +1,40 @@
...
-This is the problem Phantom is here to solve: Create an application that allows simulating bus movements and enabling passengers to track their locations & movements. 
+### Getting started with env 
 
+After cloning or rebasing on main, just create three files named: **.env.development** , **.env.production** and **.env.staging**. 
 
-GETTING STARTED
+Copy all the environment variables shared in **.env.example** file to each of the three files. Assign values to each variable. eg: 
+`PORT=5500` in  `.env.development`
+`PORT=5000` in `.env.production`
+`PORT=6000` in `.env.staging`
```
Comment (TuyizeeAnastase) [inline README.md:16]: add all this iin .env.example

File: src/index.js [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
-const swaggerJsDoc = require("swagger-jsdoc");
-const busesRouter = require("./api/routes/buses.routes");
+const express = require('express')
+const cors = require('cors')
+const morgan = require('morgan')
+const low = require('lowdb')
+const swaggerUI = require('swagger-ui-express')
+const swaggerJsDoc = require('swagger-jsdoc')
+const busesRouter = require('./api/routes/buses.routes')
+const { PORT } = require('./config/dotenv')
```
Comment (TuyizeeAnastase) [inline src/index.js:8]: configure es6 and use it

File: src/index.js [hunk 4]
```diff
@@ -1,50 +1,49 @@
...
 
-app.use(cors());
-app.use(express.json());
-app.use(morgan("dev"));
+app.use(cors())
+app.use(express.json())
+app.use(morgan('dev'))
 
-app.use("/buses", busesRouter);
+app.use('/buses', busesRouter)
```
Comment (TuyizeeAnastase) [inline src/index.js:47]: create routers folder add `index` and put bus routers from this`bus.routers.js `

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]
deployment is not working work on and add it in readme file.

**Comment by thepatrickniyo** [comment]
> deployment is not working work on and add it in readme file.

This is a concept only, if you read clearly you will get it. The variable is `PORT` and that is what is in `.env.example`

File: src/index.js (hunk 3 above)
Comment (thepatrickniyo) [reply src/index.js:8]: > configure es6 and use it

We can create a separate chore to handle that, it's completely different from this chore requirements

File: src/index.js (hunk 4 above)
Comment (thepatrickniyo) [reply src/index.js:47]: This was not done by me, and not in my chore. It is already on develop branch. It needs a separate chore

File: README.md (hunk 2 above)
Comment (thepatrickniyo) [reply README.md:16]: This is a concept if you read clearly you see that you 1 `.env` file is enough and the variable there is `PORT` port is already added in `.env.example`.

**Pull request merged**